dsc/src/util.rs

```diff
@@ -357,3 +360,28 @@ pub fn get_input(input: &Option<String>, stdin: &Option<String>, path: &Option<S
 
     parse_input_to_json(&value)
 }
+
+pub fn process_macros(json_string: &str, config_path: &str) -> String
+{
+    let path = Path::new(config_path);
+    let config_root = match path.parent()
+    {
+        Some(dir_path) => { dir_path.to_str().unwrap_or_default().to_string()},
+        _ => String::new()
+    };
+
+    // Set env var so child processes (of resources) can use it
+    debug!("Setting 'DSCConfigRoot' env var as '{}'", config_root);
+    env::set_var("DSCConfigRoot", config_root.clone());
+
+    // we are replacing a substring in json, so the new substring must be properly escaped per json rules
+    let v = json!(config_root);
+    let mut json_escaped_config_root = v.to_string();
+    json_escaped_config_root.pop(); // remove last double quote
+    if !json_escaped_config_root.is_empty() {
+        json_escaped_config_root.remove(0); // remove first double quote
+    }
+
+    debug!("Escaped json 'DSCConfigRoot' is '{}'", json_escaped_config_root);
+    json_string.replace("_DSCConfigRoot_", &json_escaped_config_root)
```

Comment:
We don't want to introduce this syntax within the config as a way to get env vars even this special one.  As noted in the issue, we eventually will want a function to retrieve any env var.

Reply:
I agree. Added `envvar` function and updated PR description.